plugin/check_extra_field_author_company/CheckExtraFieldAuthorsCompanyPlugin.php

```diff
@@ -0,0 +1,427 @@
+        }
+    }
+
+    /**
+     * Remove the extra fields "company".
+     */
+    public function removeCompanyField()
+    {
+        $data = $this->getCompanyField();
+        $this->deleteQuery($data);
+    }
+
+    /**
+     * Remove the extra fields "authors".
+     */
+    public function removeAuthorsField()
+    {
+        $data = $this->getAuthorsField();
+        $this->deleteQuery($data);
+    }
+
+    /**
+     *Insert the option fields for company with the generic values Company 1, company 2 and company 3.
+     */
+    public function setCompanyExtrafieldData()
+    {
+        $companys = [
+            0 => 'Company 1',
+            1 => 'Company 2',
+            2 => 'Company 3',
+        ];
+        $order = 0;
+        $query = "SELECT
+					id
+				FROM
+					".$this->tblExtraField."
+				WHERE
+					variable = 'company'";
+        $companyId = Database::fetch_assoc(Database::query($query));
+        $companyId = isset($companyId['id']) ? $companyId['id'] : null;
+        for ($i = 0; $i < count($companys); $i++) {
+            $order = $i + 1;
+            $extraFieldOptionValue = $companys[$i];
+            if ($companyId != null) {
+                $query = "SELECT * from ".$this->tblExtraFieldOption." where option_value = '$extraFieldOptionValue'";
+                $extraFieldOption = Database::fetch_assoc(Database::query($query));
+
+                if (isset($extraFieldOption['id']) && $extraFieldOption['id'] && $extraFieldOption['field_id'] == $companyId) {
+                    // Update?
+                } else {
+                    $query = "
+                    INSERT INTO ".$this->tblExtraFieldOption."
+                        (`field_id`, `option_value`, `display_text`, `priority`, `priority_message`, `option_order`) VALUES
+                        ( '$companyId', '$extraFieldOptionValue', '$extraFieldOptionValue', NULL, NULL, '$order');
+                    ";
+                    $data = Database::query($query);
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns the array of an element in the database that matches the variable.
+     *
+     * @param string $variableName
+     *
+     * @return array
+     */
+    protected function getInfoExtrafield($variableName = null)
+    {
+        if ($variableName == null) {
+            return [];
+        }
+        $variableName = strtolower(Database::escape_string(Security::remove_XSS($variableName)));
+        $tblExtraField = $this->tblExtraField;
+        $query = "SELECT
+			*
+		FROM
+			$tblExtraField
+		WHERE
+			variable = '$variableName'";
+        $data = Database::fetch_assoc(Database::query($query));
+        if ($data == false or !isset($data['display_text'])) {
+            return [];
+        }
+
+        return $data;
+    }
+
+    /**
+     * Executes the insertion or update of the queries in the database for the extra_field table.
+     *
+     * @param $data
+     */
+    protected function queryInsertUpdate($data)
```

Comment:
There is a method to create extrafields in `ExtraField::save`

See `tests/scripts/insert_session_fields.php` for example.